sormas-backend/src/test/java/de/symeda/sormas/backend/sormastosormas/validation/InfraValidationSoundnessTest.java

```diff
@@ -0,0 +1,581 @@
+	 * @param caption
+	 *            the caption we put on the leaf infra ref dto
+	 */
+	private void injectWrongInfra(Queue<DtoData> path, Object currentObject, String caption)
+		throws NoSuchMethodException, InvocationTargetException, InstantiationException, IllegalAccessException {
+		DtoData current = path.poll();
+
+		if (current == null) {
+			return;
+		}
+
+		// first elem is always the dto itself, skip it
+		if (current.field.getName().equals("dtoUnderTest")) {
+			current = path.poll();
+			if (current == null) {
+				return;
+			}
+		}
+
+		final ResolvedField currentField = current.field;
+		Object chillObject;
+
+		// get access to all fields of the current object
+		List<Field> availableFields = getFields(currentObject);
+
+		// given the found fields, find our target field which we want to set
+		Field injectionPoint = availableFields.stream().filter(f -> f.getName().equals(currentField.getName())).collect(Collectors.toList()).get(0);
+		injectionPoint.setAccessible(true);
+
+		// find out if the field is already initialized
+		chillObject = injectionPoint.get(currentObject);
+		if (chillObject == null) {
+			// populate the field
+
+			if (path.peek() == null) {
+				// we reached a leaf, get the InfrastructureDataReferenceDto(String uuid, String caption, String externalId) constructor
+				Constructor<?> constructor = currentField.getType().getErasedType().getConstructor(String.class, String.class, String.class);
+				chillObject = constructor.newInstance(DataHelper.createConstantUuid(0), caption, "");
+			} else {
+				// we still descend through the object tree
+				Constructor<?> constructor = currentField.getType().getErasedType().getConstructor();
+				chillObject = constructor.newInstance();
+			}
+			// assign the created object to the current field
+			injectionPoint.set(currentObject, chillObject);
+		}
+
+		// if the current field is a list we need to descend through the elements of the list, not the list object itself
+		if (currentField.getType().isInstanceOf(List.class)) {
+			List list = (List) chillObject;
+			if (list.isEmpty()) {
+				// we just created the list, create a new element for the list
+				ResolvedType elementType = currentField.getType().getTypeParameters().get(0);
+				Constructor<?> constructor = elementType.getErasedType().getConstructor();
+				chillObject = constructor.newInstance();
+				list.add(chillObject);
+			} else {
+				chillObject = list.get(0);
+			}
+		}
+
+		if (currentField.getType().isInstanceOf(Map.class)) {
+			throw new NotImplementedException();
+		}
+
+		currentObject = chillObject;
+		injectWrongInfra(path, currentObject, caption);
+
+	}
+
+	/**
+	 * Get all field of t from its class and all superclasses
+	 * 
+	 * @param t
+	 *            The object we want the fields of
+	 * @param <T>
+	 *            The type of the object
+	 * @return All fields contained in object t fof type T
+	 */
+	private <T> List<Field> getFields(T t) {
```

Comment:
we really need the Fields here as we need to access them directly. `getDeclaredField` does not include super class fields and `ResolvedField` does not link to `Field`.

Reply:
What I mean is that we could add a `fieldName` parameter an use `::getDeclaredField` instead of `getDeclaredFields`.
And also, there is `ResolvedMember::getRawMember`.

Reply:
got it fixed